### ISSUE TITLE
Can we add null images (images without any label) in training and validation set for training any model?

### ISSUE THREAD

**Issue by Haseeeb21** [ultralytics/ultralytics#16305]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Let say I am training a model on specific single class and to reduce false positives can I add extra images without any labels in it like of background or something that the model will usually see.

@glenn-jocher I have a few questions.

1. Null images can be of anything yes? like if my task is of detecting a single class or multi class but for null image I should add an image in which that class / classes are not present right?
2. Is it necessary to add labels of the same name as of images (which do not contain any class that we are training on) and if yes should it be empty (nothing in the .txt file)?
3. Where should we add null images, in train dataset, or valid dataset, or can be added in both?
4. Can we add null images in every task, such as detection, segmentation, pose, etc..? and same format to follow as question 1?
5. Does YOLOv8 / v9 / v10 automatically ignores / removes null images from the dataset during training? If yes how can I set to not remove them and train the model including null images?
6. What does adding null images benefits for the model? performance? accuracy? what other advantages?
7. Any disadvantage or risk by adding null images? except for time?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Haseeeb21, thank you for reaching out with your questions on Ultralytics 🚀!

It seems you're exploring the nuances of adding null images in your datasets. For new users, we recommend checking out the [Docs](https://docs.ultralytics.com) where you can find many helpful examples and explanations.

### If this is a 🐛 Bug Report:
Please ensure you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us understand the issue more clearly.

### For your ❓ Question:
Please provide any additional context, such as sample images or configurations, to help us assist you better. It's also helpful to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance.

For a lively discussion and real-time support, you can join our [Discord](https://ultralytics.com/discord) 🎧. For an in-depth exchange of ideas, the [Discourse](https://community.ultralytics.com) forum might be more suitable. You are also welcome to share insights and participate on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure your issue isn't related to a version mismatch, please upgrade to the latest `ultralytics` package with:

```bash
pip install -U ultralytics
```

Make sure all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are satisfied in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLOv8 runs smoothly in these verified environments (all dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) are preinstalled):

- Notebooks with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>
- [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the current passing status of our CI tests, verifying functionality across supported platforms and tasks.

An Ultralytics engineer will assist you further shortly. Thanks for being part of our community! 🌟

**Comment by Y-T-G** [comment]
1. Yes.
2. Both are correct. You may add an empty label file or you can just not add any label file and it will be considered background.
3. Up to you. If you have some images that have no objects and yet produce FPs, you can add to validation so that the metrics consider FPs on those images.
4. Yes.
5. No
6. It reduces recall slightly, while improving precision.
7. It shouldn't be a lot. Follow the guidelines [here](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/). It's called "background images" in Ultralytics docs.

**Comment by Haseeeb21** [comment]
> 1. Yes.
> 2. Both are correct. You may add an empty label file or you can just not add any label file and it will be considered background.
> 3. Up to you. If you have some images that have no objects and yet produce FPs, you can add to validation so that the metrics consider FPs on those images.
> 4. Yes.
> 5. No
> 6. It reduces recall slightly, while improving precision.
> 7. It shouldn't be a lot. Follow the guidelines [here](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/). It's called "background images" in Ultralytics docs.


Here it says the model skips the null images
https://github.com/ultralytics/ultralytics/issues/7981#issuecomment-2106412763

**Comment by Y-T-G** [comment]
@Haseeeb21 It doesn't. You can check the logs when it starts training. It will display the number of images it detected as "background".

**Comment by Haseeeb21** [comment]
> @Haseeeb21 It doesn't. You can check the logs when it starts training. It will display the number of images it detected as "background".

According to this it does, but if I add txt files it wouldn't skip.
https://github.com/ultralytics/ultralytics/issues/7981#issuecomment-2106412763

**Comment by Y-T-G** [comment]
@Haseeeb21 Well, it doesn't. And you can check it for yourself pretty easily by checking the logs like I said.

**Comment by Y-T-G** [comment]
> In the above setup, img2.jpg is your background image with no labels, _either represented by an absence_ of img2.txt or by an empty img2.txt file.

https://github.com/ultralytics/ultralytics/issues/11027#issuecomment-2089147101

**Comment by asudeeaydin** [comment]
Hello, 

I've read the thread a couple of times now. But I still feel a bit lost, I'm using a Yolov6 network and have built my own dataloader. Could you describe what's going on from a lower level as well? ie. 

- What are the bbox and class values passed in for the absence of a label (ie. a null image), are they `None` values?
- What is the loss score exactly and how a null image contributes to ie (ie. is it only through the objectiveness score). 

I've tried to use None values in my custom dataloader but soon realized that this fails in the collate_fn function since None values are not a thing in torch.

**Comment by glenn-jocher** [comment]
@asudeeaydin For null/background images at the dataloader level, the bbox and class values are **empty tensors**, not `None` values. Looking at the [verify_image_label function](https://docs.ultralytics.com/reference/data/utils/#ultralytics.data.utils.verify_image_label), when no labels are present it creates `lb = np.zeros((0, 5), dtype=np.float32)` - an empty array with shape `[0, 5]` rather than `None`.

During training, these empty label tensors allow the model to learn from pure background samples through the objectness/confidence loss component, helping reduce false positives. The loss function can handle empty ground truth tensors and will compute appropriate background losses for these images.

For your custom dataloader, ensure you're returning empty tensors (shape `[0, 5]` for detection) rather than `None` values to avoid collate_fn issues.
